apps/src/applab/designElements/library.js

```diff
@@ -56,5 +61,40 @@ module.exports = {
     element.style.top = top + 'px';
 
     return element;
+  },
+
+  getElementPropertyTable: function (elementType) {
+    return elements[elementType].PropertyTable;
+  },
+
+  /**
+   * @param {HTMLElement} element
+   * @returns {string} String representing elementType
+   */
+  getElementType: function (element) {
+    var tagname = element.tagName.toLowerCase();
+    switch (tagname) {
+      case 'button':
+        return 'BUTTON';
```

Comment:
This got me thinking about what the advantages of using an enum here would be. One reason good reason to use enums in other languages is so that you get compiler errors if you have a typo (i.e. ElementType.BUTON would give me an error). We don't get that benefit in JS.

Another benefit in other languages is nice auto completion. In many editors, i could type ElementType. and get a list of possibilities. Again, this is something I'm not going to get writing JS.

Another argument in favor of enums is it gives you a single place to find a list of the possible values. In our case, we already have that in the keys of elements.

Can you think of arguments in favor of enums in JS? I guess one reasonably compelling argument is that when I see "BUTTON", I don't have a lot of context, whereas as ElementType.BUTTON makes it clear that I'm using "BUTTON" as a type of element, and disambiguates from usages as a tagName for example. I think I've maybe just convinced myself an enum is good here :-)